cmd/shimdiag/share.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/appargs"
+	"github.com/Microsoft/hcsshim/internal/shimdiag"
+	"github.com/urfave/cli"
+)
+
+// LCOW root FS VHD is mounted read only so the only place you can mount a directory
+// is over an existing mount point like /home etc. Single file mapping will not work and
+// will return 'mkdir /path: read-only file system'. You can set the preferred root fs type
+// to initrd to circumvent this.
+
+var readOnly bool
```

Comment:
To avoid a global, you could do something like this in the Action func:
`readOnly := context.Bool(readOnlyFlagName)`

Reply:
As readonly can either be passed in as -readonly or -ro I'll have to do 
```go
readOnly := false
if ctx.Bool("ro") || ctx.Bool("readonly"){
        readOnly = true
}
```

OR just get rid of the option to pass in -readonly and just leave -ro :P

Reply:
I'll just do 
```go
readOnly := c.Bool("ro") || c.Bool("readonly")
```
actually

Reply:
I think you only need to check "readonly" see wclayer's import command as an example.

Reply:
Oh even better :) let me test this just to be sure

Reply:
Yep, it works